qualang_tools/plot/plot.py

```diff
@@ -1,17 +1,296 @@
+    _plot_options = {
+        "fontsize": 14,
+        "color": "b",
+        "marker": "o",
+        "linewidth": 0,
+        "figsize": None,
+    }
+    if plot_options is not None:
+        for key in plot_options:
+            if key in _plot_options.keys():
+                _plot_options[key] = plot_options[key]
+            else:
+                raise ValueError(
+                    f"The key '{key}' in 'plot_options' doesn't exists. The available options are {[option for option in _plot_options.keys()]}"
+                )
+
+    if amp_and_phase:
+        y1 = np.sqrt(I**2 + Q**2)
+        y2 = signal.detrend(np.unwrap(np.angle(I + 1j * Q)))
+        y1_label = "Amplitude [a.u.]"
+        y2_label = "Phase [rad]"
+    else:
+        y1 = I
+        y2 = Q
+        y1_label = "I [a.u.]"
+        y2_label = "Q [a.u.]"
+
+    if fig is None:
+        fig = plt.figure(figsize=_plot_options["figsize"])
+    plt.suptitle(title, fontsize=_plot_options["fontsize"] + 2)
+    ax1 = plt.subplot(211)
+    plt.cla()
+    plt.plot(
+        x,
+        y1,
+        color=_plot_options["color"],
+        marker=_plot_options["marker"],
+        linewidth=_plot_options["linewidth"],
+    )
+    plt.ylabel(y1_label, fontsize=_plot_options["fontsize"])
+    plt.xticks(fontsize=_plot_options["fontsize"])
+    plt.yticks(fontsize=_plot_options["fontsize"])
+    plt.subplot(212, sharex=ax1)
+    plt.cla()
+    plt.plot(
+        x,
+        y2,
+        color=_plot_options["color"],
+        marker=_plot_options["marker"],
+        linewidth=_plot_options["linewidth"],
+    )
+    plt.xlabel(x_label, fontsize=_plot_options["fontsize"])
+    plt.ylabel(y2_label, fontsize=_plot_options["fontsize"])
+    plt.xticks(fontsize=_plot_options["fontsize"])
+    plt.yticks(fontsize=_plot_options["fontsize"])
+    plt.tight_layout()
+    return fig
+
+
+def plot_channel(ax, data: np.ndarray, title: str):
+    """
+    Plot data from one channel to a given axis.
+
+    :param ax: valid matplotlib.pyplot axis .
+    :param data: 1D array containing the data to be plotted, can be real or complex.
+    :param title: title of the plot.
+
+    """
+    if data.dtype == np.complex:
+        ax.plot(data.real, label="Real part")
+        ax.plot(data.imag, label="Imaginary part")
+        ax.legend(loc="upper right")
+    else:
+        ax.plot(data)
+    ax.set_title(title)
+    ax.set_ylim((-0.6, 0.6))
+
+
+def get_simulated_samples_by_element(element_name: str, job: QmJob, config: dict):
+    """
+    Extract the simulated samples corresponding to a given element.
+
+    :param element_name: name of the element corresponding to the samples to extract. Must be defined in the config.
+    :param job: The simulated QmJob from which the samples will be extracted.
+    :param config: The config file used to create the job.
+    :return: 1D complex array containing the simulated samples for the specified element.
+    """
+    element = config["elements"][element_name]

```

Comment:
Probably need to add here a `wait_for_all_values()`

Reply:
All what happens here is meant to execute once the simulation is done, so I would prefer to put the `wait_for_all_values()` in the program rather than in the function 
Also, I am not sure that we need it since we just want to extract the simulated samples and not something from `stream_processing`, what do you think?

Reply:
The information from the simulation will also not be available until the `wait_for_all_values()` is finished.
So if you write in a code:
```python
job = qmm.simulate(...)
get_simulated_samples_by_element(...)
```
You'll get an error...

Reply:
I don't see why we need `wait_for_all_values()` here...
Maybe it depends on the versions, but I retested it and I don't have any error without it

Reply:
Did you simulate a program with stream processing? And let it run to actually get a bit of data?
You don't have access to the simulator samples until the SP is done. Trust me...
If your program didn't had it, or very simple SP, then it might work without (due to the python delay just being long enough). But in many programs it will give an error...

Reply:
Yes, but it was fairly simple (resonator spec) so I couldn't get any error, no matter the simulation duration or number of points...
But I believe you and I will add the `wait_for_all_values()` which cannot harm anyway